gobblin-temporal/src/main/java/org/apache/gobblin/temporal/yarn/YarnService.java

```diff
@@ -485,8 +448,11 @@ private void requestContainer(Optional<String> preferredNode, Resource resource)
     priority.setPriority(priorityNum);
 
     String[] preferredNodes = preferredNode.isPresent() ? new String[] {preferredNode.get()} : null;
+
+    long allocationRequestId = optAllocationRequestId.or(0L);
```

Comment:
rather than this magic number, name the constant `DEFAULT_ALLOCATION_REQUEST_ID` and then use that to also initialize the `allocationRequestIdGenerator`